module/VuFind/src/VuFind/Connection/LibGuides.php

```diff
@@ -0,0 +1,192 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Connection
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Brent Palmer <brent-palmer@icpl.org>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+
+namespace VuFind\Connection;
+
+use Exception;
+use Laminas\Log\LoggerAwareInterface;
+
+/**
+ * LibGuides API connection class.
+ *
+ * Note: This is for the LibGuides API used by the LibGuidesProfile recommendation service,
+ * this is *not* for the LibGuides search widget "API" used by the LibGuides and LibGuidesAZ
+ * data sources.
+ *
+ * Closely adapted from VuFind\DigitalContent\OverdriveConnector.
+ *
+ * @category VuFind
+ * @package  Connection
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Brent Palmer <brent-palmer@icpl.org>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class LibGuides implements
+    OauthServiceInterface,
+    \VuFindHttp\HttpServiceAwareInterface,
+    LoggerAwareInterface
+{
+    use OauthServiceTrait;
+    use \VuFindHttp\HttpServiceAwareTrait;
+    use \VuFind\Log\LoggerAwareTrait {
+        logError as error;
+    }
+
+    /**
+     * HTTP Client
+     *
+     * @var \Laminas\Http\HttpClient
+     */
+    protected $client;
+
+    /**
+     * Base URL of the LibGuides API
+     *
+     * @var string
+     */
+    protected $baseUrl;
+
+    /**
+     * Client ID for a client_credentials grant
+     *
+     * @var string
+     */
+    protected $clientId;
+
+    /**
+     * Client Secret for a client_credentials grant
+     *
+     * @var string
+     */
+    protected $clientSecret;
+
+    /**
+     * Constructor
+     *
+     * @param \Laminas\Http\Client $client             HTTP client
+     * @param string               $baseUrl            Base URL of the LibGuides API
+     * @param string               $clientId           Client Id provided by LibGuides API configuration
+     * @param string               $clientSecret       Client Secret provided by LibGuides API configuration
+     * @param bool                 $forceNewConnection Force a new connection (get a new token)
+     *
+     * @link https://ask.springshare.com/libguides/faq/873#api-auth
+     */
+    public function __construct(
+        $client,
+        $baseUrl,
+        $clientId,
+        $clientSecret,
+        $forceNewConnection = false
+    ) {
+        $this->client = $client;
+        $this->baseUrl = $baseUrl;
+        $this->clientId = $clientId;
+        $this->clientSecret = $clientSecret;
+    }
+
+    /**
+     * Load all LibGuides accounts.
+     *
+     * @return array An array of all LibGuides accounts, or an empty array
+     * if an error occurs
```

Comment:
I know returning an empty array is unusual.  It has the benefit of being useful, but if it's better to return null or throw an exception or something...?

Reply:
Yup, changed to return null so I can use null coalescing when calling it.